pkg/gardenlet/controller/controllerinstallation/controllerinstallation/reconciler_test.go

```diff
@@ -0,0 +1,147 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package controllerinstallation_test
+
+import (
+	"math"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/ptr"
+
+	. "github.com/gardener/gardener/pkg/gardenlet/controller/controllerinstallation/controllerinstallation"
+)
+
+var _ = Describe("Reconciler", func() {
+	var (
+		reconciler *Reconciler
+	)
+
+	BeforeEach(func() {
+		reconciler = &Reconciler{}
+	})
+
+	Describe("#BootstrapControlPlaneNodeFunc", func() {
+		var (
+			pod         *corev1.Pod
+			deployment  *appsv1.Deployment
+			statefulSet *appsv1.StatefulSet
+			daemonSet   *appsv1.DaemonSet
+			job         *batchv1.Job
+			cronJob     *batchv1.CronJob
+		)
+
+		BeforeEach(func() {
+			pod = &corev1.Pod{}
+			deployment = &appsv1.Deployment{}
+			statefulSet = &appsv1.StatefulSet{}
+			daemonSet = &appsv1.DaemonSet{}
+			job = &batchv1.Job{}
+			cronJob = &batchv1.CronJob{}
+		})
+
+		It("should not change objects if not bootstrapping control plane node", func() {
+			reconciler.BootstrapControlPlaneNode = false
+
+			p := pod.DeepCopy()
+			Expect(reconciler.BootstrapControlPlaneNodeFunc(p)).To(Succeed())
+			Expect(p).To(Equal(pod))
+
+			d := deployment.DeepCopy()
+			Expect(reconciler.BootstrapControlPlaneNodeFunc(d)).To(Succeed())
+			Expect(d).To(Equal(deployment))
+
+			s := statefulSet.DeepCopy()
+			Expect(reconciler.BootstrapControlPlaneNodeFunc(s)).To(Succeed())
+			Expect(s).To(Equal(statefulSet))
+
+			ds := daemonSet.DeepCopy()
+			Expect(reconciler.BootstrapControlPlaneNodeFunc(ds)).To(Succeed())
+			Expect(ds).To(Equal(daemonSet))
+
+			j := job.DeepCopy()
+			Expect(reconciler.BootstrapControlPlaneNodeFunc(j)).To(Succeed())
+			Expect(j).To(Equal(job))
+
+			c := cronJob.DeepCopy()
+			Expect(reconciler.BootstrapControlPlaneNodeFunc(c)).To(Succeed())
+			Expect(c).To(Equal(cronJob))
+		})
+
+		It("should adapt objects if bootstrapping control plane node", func() {
+			reconciler.BootstrapControlPlaneNode = true
+
+			checkHostNetworkAndTolerations := func(podSpec *corev1.PodSpec) {
+				Expect(podSpec.HostNetwork).To(BeTrue())
```

Comment:
Let's add an offset to these assertions so that the test failure indicates for which object kind an assertion failed.
We can either add an offset to the individual assertions or add `GinkgoHelper()`:
```suggestion
				GinkgoHelper()
				Expect(podSpec.HostNetwork).To(BeTrue())
```